README.md

```diff
@@ -8,43 +8,105 @@ This repo builds a Docker image which collects tooling used to setup environment
-
-apply:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) apply $(DIR) $(ENV) $(SUFFIX)
-
-destroy:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) destroy $(DIR) $(ENV) $(SUFFIX)
-
-state-remove:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) state-remove $(DIR) $(ENV) $(SUFFIX)
-
-validate:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) validate $(DIR) $(ENV) $(SUFFIX)
+AZURE_DIR_MOUNT:=$(shell echo "-v $(AZURE_CONFIG_DIR):/work/.azure")
+COMMAND:=$(shell echo "docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" --env-file $(TEMP_ENV_FILE) $(AZURE_DIR_MOUNT) -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE)")
+CLEANUP_COMMAND:=$(shell echo "$(MAKE) --no-print-directory teardown TEMP_ENV_FILE=$(TEMP_ENV_FILE)")
+
+.PHONY: setup
+.SILENT: setup
+setup:
+	set -e
+
+	mkdir -p $(AZURE_CONFIG_DIR)
+	export AZURE_CONFIG_DIR="$(AZURE_CONFIG_DIR)"
+
+	if [ "$${CI}" == "true" ]; then
+		echo ARM_CLIENT_ID=$${servicePrincipalId} >> $(TEMP_ENV_FILE)
+		echo ARM_CLIENT_SECRET=$${servicePrincipalKey} >> $(TEMP_ENV_FILE)
+		echo ARM_TENANT_ID=$${tenantId} >> $(TEMP_ENV_FILE)
+	fi
+
+	echo ARM_SUBSCRIPTION_ID=$$(az account show -o tsv --query 'id') >> $(TEMP_ENV_FILE)
+
+	if [ "$(AWS_ENABLED)" == "true" ]; then
+		if [ "$${CI}" == "true" ]; then
+			echo AWS_ACCESS_KEY_ID=$${AWS_ACCESS_KEY_ID} >> $(TEMP_ENV_FILE)
+			echo AWS_SECRET_ACCESS_KEY=$${AWS_SECRET_ACCESS_KEY} >> $(TEMP_ENV_FILE)
+			echo AWS_DEFAULT_REGION=$${AWS_DEFAULT_REGION} >> $(TEMP_ENV_FILE)
+		else
+			AWS_ROLE_ARN=$$(aws configure get role_arn)
+			AWS_VARS=$$(aws sts assume-role --role-arn $${AWS_ROLE_ARN} --role-session-name awscli --output json | grep -E "AccessKeyId|SecretAccessKey|SessionToken" | sed "s/[ |,|\"]//g")
+			echo AWS_ACCESS_KEY_ID=$$(echo "$${AWS_VARS}" | grep AccessKeyId | sed "s/AccessKeyId://g") >> $(TEMP_ENV_FILE)
+			echo AWS_SECRET_ACCESS_KEY=$$(echo "$${AWS_VARS}" | grep SecretAccessKey | sed "s/SecretAccessKey://g") >> $(TEMP_ENV_FILE)
+			echo AWS_SESSION_TOKEN=$$(echo "$${AWS_VARS}" | grep SessionToken | sed "s/SessionToken://g") >> $(TEMP_ENV_FILE)
```

Comment:
I *think* these four lines can be formulated something like:
```
aws sts assumre-role --role-arn $${AWS_ROLE_ARN} --role-session-name awscli --output text | \
    read -r AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN
cat >> $(ENV_TEMP_FILE) <<EOF
AWS_ACCESS_KEY_ID=$${AWS_ACCESS_KEY_ID}
AWS_SECRET_ACCESS_KEY=$${AWS_SECRET_ACCESS_KEY}
AWS_SESSION_TOKEN=$${AWS_SESSION_TOKEN}
EOF
```
(possibly there are more value that needs dummys). Means no grep and sed dependency.